pkg/engine/lifecycletest/test_plan.go

```diff
@@ -237,6 +265,192 @@ func (op TestOp) runWithContext(
 	return nil, snap, errors.Join(errs...)
 }
 
+// We're just checking that we have the right number of events and
+// that they have the expected types.  We don't do a deep comparison
+// here, because all that matters is that we have the same events in
+// some order.  The non-display tests are responsible for acually
+// checking the events properly.
+func compareEvents(expected, actual []engine.Event) error {
+	encountered := make(map[int]bool)
+	if len(expected) != len(actual) {
+		return fmt.Errorf("expected %d events, got %d", len(expected), len(actual))
+	}
+	for _, e := range expected {
+		found := false
+		for i, a := range actual {
+			if encountered[i] {
+				continue
+			}
+			if a.Type == e.Type {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return fmt.Errorf("expected event %v not found", e)
+		}
+	}
+	return nil
+}
+
+// TODO: dedup with diff_test.go
```

Comment:
In general, any TODO we have in the code should have an associated tracking issue and be linked to from here.

Reply:
Thanks for calling this out!  It's actually easy enough to just address it rather than leaving a TODO around, so I just did that.

Reply:
I gave that a try and it ends up with an annoying circular dependency that's not easy to fix.  Given that, and the fact that this is only used in two places, I don't think it's really worth the effort of factoring it out at this point, so I just removed the TODO.